src/components/data/DataTable.tsx

```diff
@@ -49,6 +50,10 @@ export type DataTableProps<Row> = CompositeProps &
   ComponentProps<Row> &
   Omit<JSX.HTMLAttributes<HTMLElement>, 'size' | 'rows' | 'role' | 'loading'>;
 
+function defaultRenderItem<Row>(r: Row, field: keyof Row): ComponentChildren {
+  return r[field] as ComponentChildren;
+}
```

Comment:
Hoisted this default prop value to a top-level function so it doesn't change on each render.